GSoC25/TestFiles/test_predicate_scoring.py

```
@@ -0,0 +1,309 @@
+        return entity_type.lower() in constraint_type.lower() or constraint_type.lower() in entity_type.lower()
+    
+    def compute_predicate_similarity(self, evidence_text: str, predicate_uri: str) -> float:
+        """Compute max cosine similarity between evidence and predicate description"""
+        if predicate_uri not in self.predicate_descriptions:
+            return 0.0
+            
+        predicate_desc = self.predicate_descriptions[predicate_uri]
+        
+        # Use pre-fitted vectorizer to avoid refitting
+        if self._fitted_vectorizer is None:
+            all_texts = [evidence_text, predicate_desc] + list(self.predicate_descriptions.values())
```

Ruff (0.14.2)

Consider iterable unpacking instead of concatenation

Replace with iterable unpacking

(RUF005)